crates/turbopack-ecmascript/src/chunk/mod.rs

```diff
@@ -512,10 +512,12 @@ async fn module_factory(content: EcmascriptChunkItemContentVc) -> Result<CodeVc>
         "c: __turbopack_cache__",
         "l: __turbopack_load__",
         "p: process",
-        "g: global",
         // HACK
         "__dirname",
     ];
+    if content.options.global {
+        args.push("g: global");
+    }
     if content.options.module {
```

Comment:
We need `global` unconditionally for now, as the random code can also use `global` and we don't detect that in analysis for now.

Reply:
In future the condition is a good idea.

Reply:
Reverted.